main.go

```diff
@@ -401,6 +401,13 @@ func anchorIDForLocalType(t *types.Type, typePkgMap map[*types.Type]*apiPackage)
 func linkForType(t *types.Type, c generatorConfig, typePkgMap map[*types.Type]*apiPackage) (string, error) {
 	t = tryDereference(t) // dereference kind=Pointer
 
+	if t.Kind == types.Pointer {
```

Comment:
can't we just tryDereference it in a loop until it is not Pointer anymore? there is no guarantee that 1-level or 2-level would solve it magically as you said.

Reply:
I also recommend you to evaluate all places we call `tryDereference` and see if they are also susceptible to this.

maybe we need something like `tryDereferencePointer` method and use it all over the place?

Reply:
Thank you @ahmetb You are right. Let me work upon these aspects and update the PR accordingly.